launcher/settingsView/configeditordialog_moc.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * configeditordialog_moc.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#include "StdInc.h"
+#include "configeditordialog_moc.h"
+#include "ui_configeditordialog_moc.h"
+
+#include "../helper.h"
+#include "../mainwindow_moc.h"
+
+#include "../../lib/json/JsonNode.h"
+#include "../../lib/VCMIDirs.h"
+#include "../../lib/json/JsonFormatException.h"
+
+ConfigEditorDialog::ConfigEditorDialog(QWidget *parent):
+	QDialog(parent),
+	ui(new Ui::ConfigEditorDialog)
+{
+	ui->setupUi(this);
+
+	setWindowTitle(tr("Config editor"));
+	
+	setWindowModality(Qt::ApplicationModal);
+#ifdef VCMI_MOBILE
+	showMaximized();
+#else
+	show();
+#endif
+
+	QStringList files = {
+		"settings.json",
+		"persistentStorage.json",
+		"modSettings.json"
+	};
+	ui->comboBox->addItems(files);
+
+	QFont font("Monospace");
+	font.setStyleHint(QFont::TypeWriter);
+	ui->plainTextEdit->setFont(font);
+	ui->plainTextEdit->setLineWrapMode(QPlainTextEdit::LineWrapMode::NoWrap);
+	ui->plainTextEdit->setPlainText(loadFile(files.first()));
+	
+	connect(ui->comboBox, &QComboBox::currentTextChanged, this, &ConfigEditorDialog::onComboBoxTextChanged);
+	connect(ui->closeButton, &QPushButton::clicked, this, &ConfigEditorDialog::onCloseButtonClicked);
+	connect(ui->saveButton, &QPushButton::clicked, this, &ConfigEditorDialog::onSaveButtonClicked);
+	connect(ui->plainTextEdit, &QPlainTextEdit::cursorPositionChanged, this, [this](){
+		ui->plainTextEdit->ensureCursorVisible();
+		ui->plainTextEdit->setTextCursor(ui->plainTextEdit->textCursor());
+	});
+}
+
+ConfigEditorDialog::~ConfigEditorDialog()
+{
+	delete ui;
+}
+
+void ConfigEditorDialog::showConfigEditorDialog()
+{
+	auto * dialog = new ConfigEditorDialog();
+	
+	dialog->setAttribute(Qt::WA_DeleteOnClose);
+}
+
+bool askUnsavedChanges(QWidget *parent)
+{
+	auto reply = QMessageBox::question(parent, QObject::tr("Unsaved changes"), QObject::tr("Do you want to discard changes?"), QMessageBox::Yes | QMessageBox::No);
+	return reply != QMessageBox::Yes;
+}
+
+void ConfigEditorDialog::onComboBoxTextChanged(const QString& text)
```

Comment:
```suggestion
void ConfigEditorDialog::onComboBoxTextChanged(QString text)
```

I guess the param should be named `filename`?